frontend/src/TestEdit.js

```diff
@@ -0,0 +1,263 @@
+            if(a > +sIndex){
+                thisArr[index]-=1;
+            }
+        });
+
+    }
+
+
+    async function addAnswer(index) {
+        let newItem = {...item};
+
+        (newItem.questions[index]).answerVariants.push([]);
+        setItem(newItem);
+    }
+
+
+    async function addQuestion() {
+        let newItem = {...item};
+        const que= {name:"", id:+(++(maxId)), answerVariants:[""],answers:[]};
+
+        newItem.questions.push(que);
+        newItem.added.add(que.id)
+        setItem(newItem);
+    }
+
+
+    function isChecked(id, sIndex){
+        return (item.questions.find(q => q.id === +id)).answers.includes(sIndex);
+    }
+
+
+    async function delQuestion(index) {
+        let newItem = {...item};
+        let questionId=newItem.questions[index].id
+
+        if(newItem.added.has(questionId)){
+            newItem.added.delete(questionId);
+        }else{
+            newItem.deleted.add(questionId);
+        }
+        newItem.questions=  newItem.questions.filter(i => i !== newItem.questions[index]);
+        setItem(newItem);
+    }
+
+
+    useEffect(() => {
+        const dataFetch = async () => {
+            const data = await (
+                await fetch(`/subload/tests/${id}`)
+            ).json();
+
+            data.questions.forEach(function (q) {
+                if (q.id > maxId) {
+                    maxId = q.id;
+                }
+            });
+
+            data.added = new Set();
+            data.deleted = new Set();
+            data.groupName = data.test.group.name;
+            setItem(data);
+        };
+
+        if (id !== 'new') {
+            dataFetch();
+        }
+    }, []);
+
+
+        return <div>
+            <AppNavbar/>
+            <Container>
+                {<h2>{item.test.id ? 'Edit Test' : 'Add Test'}</h2>}
+                <Form onSubmit={handleSubmit} className="container-fluid">
+                    <FormGroup>
+                        <Label for="name">Name</Label>
+                        <Input type="text" name="name" id="name" value={item.test.name || ''}
+                               onChange={handleChange} autoComplete="name"/>
+                    </FormGroup>
+
+                    <FormGroup>
+                        <Label for="group">For group</Label>
+                        <Input type="text" name="group" id="group"  value={item.groupName || ''}
+                               onChange={handleGroupChange}/>
+                    </FormGroup>
+
+                    <FormGroup>
+                        <Label for="duration">passing time(minutes)</Label>
+                        <Input type="number" name="duration" id="duration" value={item.test.duration || ''}
+                               onChange={handleChange} autoComplete="duration"/>
+                    </FormGroup>
+
+                    <hr></hr>
+                    {
+                        item.questions.map((form, index) => {
```

Comment:
you can use a "one-line" version of lambda (without curly braces and `return` keyword) here and across all the file

Reply:
Done